server/http/interceptors/interceptors.go

```diff
@@ -45,26 +45,35 @@ var (
 
 const contentSecurityPolicyReportingEndpointName = "csp-endpoint"
 
-var contentSecurityPolicyTemplate = strings.Join([]string{
-	"default-src 'self'",
-	"style-src 'self' https://fonts.googleapis.com/css",
-	// libsodium.js requires 'wasm-unsafe-eval' to avoid a fallback to asm.js.
-	"script-src 'self' 'strict-dynamic' 'nonce-%s' 'wasm-unsafe-eval'",
-	// Monaco editor dynamically loads fonts from its CDN.
-	"font-src 'self' https://fonts.gstatic.com https://cdnjs.cloudflare.com/ajax/libs/monaco-editor/",
-	// We directly embed profile images from Google accounts and don't control their URLs.
-	"img-src 'self' https:",
-	// libsodium.js requires data: for wasm.
-	"connect-src 'self' https://www.googletagmanager.com data:",
-	"form-action 'self'",
-	"frame-src 'none'",
-	"worker-src 'none'",
-	"frame-ancestors 'none'",
-	"base-uri 'none'",
-	"block-all-mixed-content",
-	"report-to " + contentSecurityPolicyReportingEndpointName,
-	"report-uri " + csp.ReportingEndpoint,
-}, ";")
+var contentSecurityPolicyTemplate = createContentSecurityTemplate()
+
+func createContentSecurityTemplate() string {
+	var regionConnectSrcs []string
+	for _, r := range region.Protos() {
+		regionConnectSrcs = append(regionConnectSrcs, r.Subdomains)
+	}
+	return strings.Join([]string{
+		"default-src 'self'",
+		"style-src 'self' https://fonts.googleapis.com/css",
+		// Monaco editor dynamically loads fonts from its CDN.
+		"font-src 'self' https://fonts.gstatic.com https://cdnjs.cloudflare.com/ajax/libs/monaco-editor/",
+		// We directly embed profile images from Google accounts and don't control their URLs.
+		"img-src 'self' https:",
+		"form-action 'self'",
+		"frame-src 'none'",
+		"worker-src 'none'",
+		"frame-ancestors 'none'",
+		"base-uri 'none'",
+		"block-all-mixed-content",
+		// libsodium.js requires data: for wasm.
+		"connect-src 'self' data: https://*.google-analytics.com https://*.analytics.google.com https://*.googletagmanager.com " + strings.Join(regionConnectSrcs, " "),
+		"report-to " + contentSecurityPolicyReportingEndpointName,
+		"report-uri " + csp.ReportingEndpoint,
+		// libsodium.js requires 'wasm-unsafe-eval' to avoid a fallback to asm.js.
+		// KEEP THIS LAST as setContentSecurityPolicy appends the nonce.
+		"script-src 'self' 'strict-dynamic' 'wasm-unsafe-eval'",
```

Comment:
nit: I think it's fine to pass the nonce in as a param to this function and call it for each request - we don't get a high volume of requests for the root UI endpoint (`/`) so the cost of rebuilding this string for each request should be negligible, and is worth the added simplicity IMO.

Reply:
Yeah, it should be negligible - we can always profile later and optimize if needed, but I wouldn't expect this code path to show up in a CPU profile